code/hsec-tools/src/Security/Advisories/Parse.hs

```diff
@@ -417,9 +418,10 @@ onKey ::
 onKey tbl k absent present =
   maybe absent present $ Map.lookup k tbl
 
-isInt :: TOML.Value -> TableParser Integer
-isInt (TOML.Integer i) = pure i
-isInt other = throwError $ InvalidFormat "Integer" (describeValue other) "42"
+isCWE :: TOML.Value -> TableParser CWE.CWEID
+isCWE (TOML.Integer i) | Just cwe <- CWE.mkCWEID i, Map.member cwe CWE.cweNames = pure cwe
+isCWE (TOML.String txt) | Just cwe <- Map.lookup txt CWE.cweIds = pure cwe
+isCWE other = throwError $ InvalidFormat "Integer" (describeValue other) "42"
```

Comment:
I think we should continue using the CWE number in the TOML, with optional description text, e.g.:

```
cwe = [
  "328: Use of weak hash",
  400,
]
```

If the summary text is given we should ensure it matches the CWE number (and fail parse if it does not).  But I prefer to leave the CWE ID in the TOML because it makes it easier for humans to look up references.

We could also consider accepting the bare CWE ID as a string, e.g. `"400"`, as a convenience.